avm/src/vm/opcode.rs

```diff
@@ -2758,48 +2758,77 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
     })
   });
 
-  async fn __httpget(uri: String) -> std::result::Result<String, Box<dyn std::error::Error>> {
-    let url = TryInto::<Uri>::try_into(uri)?;
-    let mut response = Client::builder()
+  fn __httpreq(
+    method: String,
+    uri: String,
+    headers: Vec<(String, String)>,
+    body: Option<String>,
+  ) -> ResponseFuture {
+    let mut req = Request::builder()
+      .method(method.as_str())
+      .uri(uri.as_str());
+    for header in headers {
+      req = req.header(header.0.as_str(), header.1.as_str());
+    }
+    let req_obj = if body.is_some() {
+      req.body(Body::from(body.unwrap())).unwrap()
+    } else {
+      req.body(Body::empty()).unwrap()
+    };
+    Client::builder()
       .build::<_, Body>(HttpsConnector::new())
-      .get(url)
-      .await?;
-    let body = hyper::body::to_bytes(response.body_mut()).await?;
-    let res = String::from_utf8(body.to_vec())?;
-    Ok(res)
+      .request(req_obj)
   }
-  io!(httpget => fn(args, mut hand_mem) {
+  io!(httpreq => fn(args, mut hand_mem) {
     Box::pin(async move {
-      let res = __httpget(HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[0]))).await;
-      hand_mem.init_fractal(args[2]);
-      hand_mem.push_fixed(args[2], if res.is_ok() { 1i64 } else { 0i64 });
-      match res {
-        Ok(res) => hand_mem.push_fractal(args[2], HandlerMemory::str_to_fractal(&res)),
-        Err(ee) => hand_mem.push_fractal(args[2], HandlerMemory::str_to_fractal(format!("{}", ee).as_str())),
+      let req = hand_mem.read_fractal(args[0]);
+      let method = HandlerMemory::fractal_to_string(hand_mem.read_from_fractal(&req, 0).0);
+      let url = HandlerMemory::fractal_to_string(hand_mem.read_from_fractal(&req, 1).0);
+      let headers = hand_mem.read_from_fractal(&req, 2).0;
+      let mut out_headers = Vec::new();
+      for i in 0..out_headers.len() {
+        let header = hand_mem.read_from_fractal(&headers, i).0;
+        let key = HandlerMemory::fractal_to_string(hand_mem.read_from_fractal(&header, 0).0);
+        let val = HandlerMemory::fractal_to_string(hand_mem.read_from_fractal(&header, 1).0);
+        out_headers.push((key, val));
       }
```

Comment:
this for loop doesn't do anything

Reply:
Oops! And it worked because I wasn't sending anything with (real) headers.